### ISSUE TITLE
Keep input stream open

### ISSUE THREAD

**Issue by oliverClimbs** [doyaaaaaken/kotlin-csv#102]
I realize that generally it doesn't make sense to keep an input stream open when the end of the stream has been reached, but I'd like to read multiple CSV files that are inside a zip file (without extracting them first).  However, that's currently not possible since the input stream gets closed when the end of the current zip entry is reached.

**Comment by doyaaaaaken** [comment]
@oliverClimbs
 Hi, thanks for your useful feedback!
I think it's would be good to introduce the `withoutClosingInputStream: Boolean` option on the CSV reader.
https://github.com/doyaaaaaken/kotlin-csv#customize
This option's default value is false, and if true is set the input stream keep open (and you need to close it manually).

Does this idea solve your problem?

**Comment by oliverClimbs** [comment]
Yes, that would be a good solution.Oliver ---- On Mon, 11 Jul 2022 03:35:48 -0700  ***@***.***  wrote ----
@oliverClimbs
Hi, thanks for your useful feedback!
I think it's would be good to introduce the withoutClosingInputStream: Boolean option on the CSV reader.
https://github.com/doyaaaaaken/kotlin-csv#customize
This option's default value is false, and if true is set the input stream keep open (and you need to close it manually).
Does this idea solve your problem?

**Comment by jsoizo** [comment]
Resolved in v2.0.0.

v2 splits the JVM overloads by ownership semantics:
- `readFromFile(File)` / `writeToFile(File)` — the library opens and closes the stream.
- `read(stream)` / `write(..., stream)` — caller-owned; the library does **not** close the stream (the writer flushes but doesn't close).

So instead of a `withoutClosingInputStream` flag, you choose by which overload you call:

```kotlin
// keep the stream open
myInputStream.use { stream ->
    csvReader().readAll(stream)  // stream stays open here
    // ...keep using stream...
}
```
